core/script.runtime/src/kotlin/script/templates/annotations.kt

```diff
@@ -22,13 +22,28 @@ import kotlin.reflect.KClass
 import kotlin.script.dependencies.DependenciesResolver.NoDependencies
 import kotlin.script.dependencies.ScriptDependenciesResolver
 
+// discuss
+
+// One of the requests is to allow to share compiler args via script templates: https://youtrack.jetbrains.com/issue/KT-19120
+// Making it a property of ScriptTemplateDefinition is a viable options but it doesn't allow to changes compiler arguments based on script content.
+// For example, it could be possible to allow file level annotation and use it like:
+// @file: ExtraCompilerArgs("-enableWhatever")
```

Comment:
I think we should think ahead if it makes sense to support this. Because if we provide only the simplest API right now and decide to add 'content-based' API later we'll end up with two APIs that do the same thing and have problems merging their results later. 
The snippet I came up with just illustrutates that the user can potentially want to decide to change compiler arguments based on ScriptContents with some custom logic.

API proposal: expand `ScriptDependencies` class with `val extraCompilerArguments: List<String> = emptyList()` so this can managed uniformly with other scipt content depent stuff.